# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/nsQuoteList.h
# Commit: e25543fae7ee
# Full Hash: e25543fae7ee03ac57b9cfbfeb2418c9e2f490fc
# Author: Martin Robinson <mrobinson@igalia.com>
# Date: 2022-06-23 03:54:50
# Regressor Bug: 1463600
# File Overlap Count: 1
# Description:
#   Bug 1463600 - Implement CSS 'contain: style' r=emilio
#   
#   Add an implementation of CSS `contain: style`. This introduces two new
#   data structures, the ContainStyleScope and ContainStyleScopeManager.
#   
# ==============================================================================

diff -r 4230bdcf07ae -r e25543fae7ee layout/base/nsQuoteList.h
--- a/layout/base/nsQuoteList.h	Wed Jun 22 16:15:57 2022 +0000
+++ b/layout/base/nsQuoteList.h	Wed Jun 22 16:16:59 2022 +0000
@@ -12,6 +12,12 @@
 #include "mozilla/Attributes.h"
 #include "nsGenConList.h"
 
+namespace mozilla {
+
+class ContainStyleScope;
+
+}  // namespace mozilla
+
 struct nsQuoteNode : public nsGenConNode {
   // open-quote, close-quote, no-open-quote, or no-close-quote
   const StyleContentType mType;
@@ -69,6 +75,8 @@
   }
 
  public:
+  explicit nsQuoteList(mozilla::ContainStyleScope* aScope) : mScope(aScope) {}
+
   // assign the correct |mDepthBefore| value to a node that has been inserted
   // Should be called immediately after calling |Insert|.
   void Calc(nsQuoteNode* aNode);
@@ -84,6 +92,9 @@
 #ifdef DEBUG
   void PrintChain();
 #endif
+
+ private:
+  mozilla::ContainStyleScope* mScope;
 };
 
 #endif /* nsQuoteList_h___ */